File: client/init-shadcn.py
import subprocess
import sys

# Use shell=True to let the shell find npx in the PATH
proc = subprocess.Popen(
    ['npx', 'shadcn@latest', 'init'],
    stdin=subprocess.PIPE,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    text=True,
    shell=True
)
# Send two Enter keys (for component library and preset)
stdout, stderr = proc.communicate(input='\n\n')
print('stdout:', stdout)
print('stderr:', stderr)
print('return code:', proc.returncode)

File: client/init-shadcn.js
const { spawn } = require('child_process');
const shadcn = spawn('/c/Program Files/nodejs/npx', ['shadcn@latest', 'init'], { stdio: ['pipe', 'pipe', 'pipe'] });
shadcn.stdin.write('\n');
shadcn.stdin.end();
shadcn.stdout.on('data', (data) => {
  console.log(`stdout: ${data}`);
});
shadcn.stderr.on('data', (data) => {
  console.error(`stderr: ${data}`);
});
shadcn.on('close', (code) => {
  console.log(`child process exited with code ${code}`);
});
